File: service/src/main/java/hu/bsstudio/race/speed/timer/configuration/SpeedTimerServiceConfiguration.java
package hu.bsstudio.race.speed.timer.configuration;

import hu.bsstudio.race.speed.timer.DefaultSpeedTimerService;
import hu.bsstudio.race.speed.timer.SpeedTimerService;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class SpeedTimerServiceConfiguration {

    @Bean
    public SpeedTimerService speedTimerService() {
        return new DefaultSpeedTimerService();
    }
}

File: server/service/src/main/kotlin/hu/bsstudio/robonaut/race/speed/timer/configuration/SpeedTimerServiceConfiguration.kt
package hu.bsstudio.robonaut.race.speed.timer.configuration

import hu.bsstudio.robonaut.race.speed.timer.BroadcastingSpeedTimerService
import hu.bsstudio.robonaut.race.speed.timer.DefaultSpeedTimerService
import hu.bsstudio.robonaut.race.speed.timer.SpeedTimerService
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class SpeedTimerServiceConfiguration(
    private val rabbitTemplate: RabbitTemplate,
) {
    @Bean
    fun speedTimerService(defaultSpeedTimerService: SpeedTimerService): SpeedTimerService =
        BroadcastingSpeedTimerService(rabbitTemplate, defaultSpeedTimerService)

    @Bean
    fun defaultSpeedTimerService(): SpeedTimerService = DefaultSpeedTimerService()
}
